recipes_source/recipes/module_load_state_dict_tips.py

```diff
@@ -0,0 +1,107 @@
+"""
+
+Tips for Loading an `nn.Module` from a Checkpoint
+=================================================
+
+In this tutorial, we will share some tips for loading a model from a checkpoint.
+In particular, we will discuss
+
+
+1.  The `mmap` keyword argument on `torch.load`
+2.  The `torch.device()` context manager
+3.  The `assign` keyword argument on `nn.Module.load_state_dict()`
+
+The following snippet of code illustrates the use of the above three utilities.
+"""
+
+import torch
+
+state_dict = torch.load('checkpoint.pth', mmap=True)
+with torch.device('meta'):
+  meta_m = SomeModule()
+meta_m.load_state_dict(state_dict, assign=True)
+
+###############################################################################
+# Taking a step back, let us inspect the following more vanilla code snippet
+# that does not use any of the features listed above
+
+state_dict = torch.load('checkpoint.pth')
+m = SomeModule()
+m.load_state_dict(state_dict)
+
+###############################################################################
+# At `torch.save` time, tensor storages are tagged with the device they are
+# saved on. At `torch.load` time, tensor storages will be loaded to the device
+# they were tagged with (unless this behavior is overridden using the
+# `map_location` flag). For ease of explanation, let us assume that the tensors
+# were saved on CPU. This means that on the first line
+
+state_dict = torch.load('checkpoint.pth')
+
+###############################################################################
+# all tensor storages will be loaded into CPU RAM, which can be problematic when
+#     1. CPU RAM is smaller than the size of the checkpoint
+#     2. The user does not want to wait for the entire checkpoint to be loaded
```

Comment:
Should I include an example of 1 by 1 processing with actual timings of this step? 

Or, as a broader question, do we want to include timing code for all of these?

Reply:
nit: this is the only place where you use "The user". most likely should be rephrased to match the rest.

Adding timing code should be simple since this code is ran to generate the tutorial. So just use time() and print :)